### ISSUE TITLE
Test suite failing due to change in default dtype when creating sparse matrices

### ISSUE THREAD

**Issue by rossbar** [networkx/networkx#4626]
As seen in the CI runs of #4624 

Not sure what the root cause is yet, but it appears to be tied to the scipy 1.6.1 release:

```python
>>> import scipy
>>> scipy.__version__
'1.6.0'
>>> import networkx as nx
>>> G = nx.Graph([(1, 1)])
>>> A = nx.adjacency_matrix(G)
>>> A
<1x1 sparse matrix of type '<class 'numpy.int64'>'
        with 1 stored elements in Compressed Sparse Row format>
```

```python
>>> import scipy
>>> scipy.__version__
'1.6.1'
>>> import networkx as nx
>>> G = nx.Graph([(1, 1)])
>>> A = nx.adjacency_matrix(G)
>>> A
<1x1 sparse matrix of type '<class 'numpy.float64'>'
        with 1 stored elements in Compressed Sparse Row format>
```

Note the change in dtype from int64 to float64.

**Comment by rossbar** [comment]
See scipy/scipy#13585.

**Comment by rossbar** [comment]
Note that scipy 1.6.2 has been release with a fix for this issue. I'm going to close this as resolved by #4628 - blocklisting 1.6.1 should remain in the requirements until the major version is bumped again.
